hyperdome_client/threads.py

```diff
@@ -235,6 +237,24 @@ def run(self):
         except:
             self.signals.error.emit("you're stuck here now")
 
+class PollForConnectedGuestTask(QtCore.QRunnable):
+    """
+    ask server if a guest has requested to connect yet
+    and return the public key when they have
+    """
+    signals = TaskSignals()
+
+    def __init__(self, session: requests.Session, server: Server, uid: str):
+        super(PollForConnectedGuestTask, self).__init__()
+        self.task = functools.partial(get_guest_pub_key, server, session, uid)
+
+    def run(self):
+        try:
+            self.signals.success.emit(self.task())
+        except:
+            self.signals.error.emit("problem getting guest pubkey")
```

Comment:
I'm assuming this includes the trace, right? If not, you probably should

Reply:
It does not, this line is essentially just sending a callback to the main thread with the string. The whole threads.py is a debugging time-bomb.

Will be fixed in a future release